insights/core/spec_factory.py

```diff
@@ -291,7 +290,11 @@ def load(self):
             return out
 
         with safe_open(self.path, "r", encoding=encoding, errors="surrogateescape") as f:
-            return [l.rstrip("\n") for l in f]
+            valid_content = [l.rstrip("\n") for l in f][-MAX_CONTENT_LINES:]
```

Comment:
According to this line, all lines of one file will be loaded into memory. For big file (more than 1GB), it will lead to out of memory.   For the command tail,  It does this without reading the entire file first, so it doesn't matter how large the file is.

Reply:
re-do it by loading the last 200MB part of files larger then 200MB.